Parsers/ASimAuthentication/Parsers/vimAuthenticationCrowdStrikeFalconHost.yaml

```diff
@@ -0,0 +1,125 @@
+Parser:
+  Title: ASIM Authentication parser for CrowdStrikeFalconHost
+  Version: '0.1.0'
+  LastUpdated: Sep 13 2023
+Product:
+  Name: CrowdStrike Falcon Endpoint Protection
+Normalization:
+  Schema: Authentication
+  Version: '0.1.3'
+References:
+- Title: ASIM Authentication Schema
+  Link: https://aka.ms/ASimAuthenticationDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
+- Title: CrowdStrike Falcon Endpoint Protection Documentation
+  Link: 
+    https://falcon.us-2.crowdstrike.com/documentation/page/d88d9ed6/streaming-api-event-dictionary
+    https://falcon.us-2.crowdstrike.com/documentation/page/eb1587d1/siem-connector#mc98af8f
+Description: |
+  This ASIM parser supports normalizing CrowdStrike Falcon Endpoint Protection logs to the ASIM Authentication normalized schema. These events are captured through CrowdStrike Falcon Endpoint Protection data connector which allows you to easily connect your CrowdStrike Falcon Event Stream with Microsoft Sentinel.
+ParserName: vimAuthenticationCrowdStrikeFalconHost
+EquivalentBuiltInParser: _Im_Authentication_CrowdStrikeFalconHost
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+  - Name: starttime
+    Type: datetime
+    Default: datetime(null)
+  - Name: endtime
+    Type: datetime
+    Default: datetime(null)
+  - Name: targetusername_has
+    Type: string
+    Default: '*'
+ParserQuery: |
+  let EventSeverityLookup = datatable (LogSeverity: string, EventSeverity: string)
+  [
+      "0", "Informational",
+      "1", "Informational",
+      "2", "Low",
+      "3", "Medium",
+      "4", "High",
+      "5", "High"
+  ];
+  let parser = (disabled: bool = false, starttime: datetime=datetime(null), endtime: datetime=datetime(null), targetusername_has: string='*') {
+      CommonSecurityLog
+      | where not(disabled)
+          and (isnull(starttime) or TimeGenerated >= starttime)
+          and (isnull(endtime) or TimeGenerated <= endtime)
+          and (DeviceVendor == "CrowdStrike" and DeviceProduct == "FalconHost")
+          and (DeviceEventCategory == "AuthActivityAuditEvent" and DeviceEventClassID in ("userAuthenticate", "twoFactorAuthenticate"))
+          and (targetusername_has == "*" or DestinationUserName has targetusername_has)
+      | lookup EventSeverityLookup on LogSeverity
+      | extend
+          EventResult = iff(EventOutcome == "true", "Success", "Failure"),
+          EventStartTime = todatetime(DeviceCustomDate1),
+          ThreatField = "SrcIpAddr",
+          ThreatOriginalConfidence= ThreatConfidence,
+          ThreatConfidence = toint(ThreatConfidence),
+          ThreatOriginalRiskLevel = tostring(ThreatSeverity)
+      | extend
+          EventCount = int(1),
+          EventSchema = "Authentication",
+          EventSchemaVersion = "0.1.3",
+          EventType = "Logon"
+      | project-rename 
+          EventProduct = DeviceProduct,
+          EventVendor = DeviceVendor,
+          SrcIpAddr = DestinationTranslatedAddress,
```

Comment:
Mapping is according to the configuration as IpAddr coming for Audit & Auth Activity is the IP of the agent from where login activity is performed.

Reply:
@niralishah-crest 
We should set DvcIpAddr and TargetIpAddr in this case. SrcIpAddr only incase Target system is different.

Reply:
Done

Reply:
@niralishah-crest @jayeshprajapaticrest my bad for Dvc instruction here. For authentication events, device fields refer to the system reporting the event. I think we can map Computer with DvcHostname field.

Reply:
@niralishah-crest @jayeshprajapaticrest 
Please resolve if taken care.

Reply:
In case of CrowdStrike, Computer field contains the IpAddr value where we have installed SIEM Connector of CrowdStrike which forwards CrowdStrike events to Sentinel. So Computer field does not indicate a device from which events are reported to CrowdStrike.